cloud/pkg/devicecontroller/controller/upstream.go

```diff
@@ -136,17 +136,32 @@ func (uc *UpstreamController) updateDeviceStatus() {
 			deviceStatus := &DeviceStatus{Status: cacheDevice.Status}
 			for twinName, twin := range msgTwin.Twin {
```

Comment:
The code indentation is too deep, you can abstract a function to returns a reference to a device status twin property, and then processing.
```golang
for twinName, twin := range msgTwin.Twin {
    statusTwin := findStatusTwinProperty(twinName, deviceStatus.Status.Twins)
    if twin.Actual != nil && twin.Actual.Value != nil {
        // set statusTwin ...
    }
    if twin.Expected != nil && twin.Expected.Value != nil {
        // set statusTwin ...
    }
}

func findStatusTwinProperty(...) *Twin {
    // todo ...
}
```

Reply:
ok,I have modified.